### ISSUE TITLE
Add LeakSanitizer support in report-converter

### ISSUE THREAD

**Issue by OmniTroid** [Ericsson/codechecker#3321]
**Describe the solution you would like**
Add support for LeakSanitizer reports in report-converter, so running with -t lsan finds and converts the leaks to plists. Lsan is described in more detail here: https://clang.llvm.org/docs/LeakSanitizer.html

**Additional context**
We're using codechecker to sanitize our codebase, and many of the issues are detected by LeakSanitizer, but we currently can't convert them to codechecker reports (report-converter does not find the LeakSanitizer entries in the logs)

**Comment by csordasmarton** [comment]
@OmniTroid This can be implemented similarly as the ub sanitizer is implemented: https://github.com/Ericsson/codechecker/tree/master/tools/report-converter/codechecker_report_converter/sanitizers/ub.

Unfortunately we are very busy in these days, so if you or somebody in your team has some free time you can help us to implement this and if you are stucked, we can help you.

**Comment by jay24rajput** [comment]
@OmniTroid Can you share the sample output of the leak sanitizer? Or probably steps to reproduce it on Linux Kernel. 

I would like to get my hands on this one 😄

**Comment by csordasmarton** [comment]
@jay24rajput I think the output is the same as in the official site what @OmniTroid linked (https://clang.llvm.org/docs/LeakSanitizer.html). So I think we can start with it:
```
==23646==ERROR: LeakSanitizer: detected memory leaks
Direct leak of 7 byte(s) in 1 object(s) allocated from:
    #0 0x4af01b in __interceptor_malloc /projects/compiler-rt/lib/asan/asan_malloc_linux.cc:52:3
    #1 0x4da26a in main memory-leak.c:4:7
    #2 0x7f076fd9cec4 in __libc_start_main libc-start.c:287
SUMMARY: AddressSanitizer: 7 byte(s) leaked in 1 allocation(s)
```

Also as far as I see this output is very similar to the memory sanitizer output, so you can implement it on the similar way as the memory sanitizer parser implemented: https://github.com/Ericsson/codechecker/tree/master/tools/report-converter/codechecker_report_converter/sanitizers/memory.

**Comment by jay24rajput** [comment]
Hey @csordasmarton 

Sorry for being late on this issue. I started developing the report-converter, I am attaching the git branch here as well. But when I run `make package` in `tools/report-converter` the following error is displayed:
```sh
# Install package in 'development mode'.
python3 -m pip install -e . -e ../codechecker_report_hash/
Obtaining file:///home/jay/codechecker/tools/report-converter
Obtaining file:///home/jay/codechecker/tools/codechecker_report_hash
ERROR: Could not find a version that satisfies the requirement codechecker_report_hash (from report-converter==0.1.0) (from versions: none)
ERROR: No matching distribution found for codechecker_report_hash (from report-converter==0.1.0)
make: *** [Makefile:43: package] Error 1
```

Can you please tell me what is going wrong here?

Github Branch: https://github.com/jay24rajput/codechecker/tree/leak_sanitizer

**Comment by csordasmarton** [comment]
@jay24rajput What is the output of this command: `python3 -m pip -V`. My pip version is `pip 21.0.1` and I didn't get any error during the make package target. Try to update your pip version, I think that will solve this problem: `python3 -m pip install --upgrade pip`

**Comment by jay24rajput** [comment]
Thanks, @csordasmarton. That helped solve the error 👍🏻
